database/tbcd/database.go

```diff
@@ -42,6 +42,36 @@ func (it InsertType) String() string {
 	return itStrings[it]
 }
 
+// Canonical chain geometry changes resulting from header removal
+type RemoveType int
+
+const (
+	RTInvalid      RemoveType = 0 // Invalid removal for generic reason (ex: no headers to remove)
+	RTChainDescend RemoveType = 1 // Removal walked the canonical chain backwards, but existing chain is still canonical
+	RTForkDescend  RemoveType = 2 // Removal walked a non-canonical chain backwards, no change to canonical chain remaining canonical
+	RTChainFork    RemoveType = 3 // Removal walked canonical chain backwards far enough that another chain is now canonical
+)
+
```

Comment:
Why do we need to remove headers? So long the headers are inserted consecutively (and if they don't bad things happen!!!) the heaviest will always win. I am not necessarily opposed but this seems like a lot of extra code that may complicate things quite a bit.